src/driver/database-connection.ts

```diff
@@ -26,6 +26,11 @@ export interface QueryResult<O> {
    */
   readonly numAffectedRows?: bigint
 
+  /**
+   * A MySQL-only field contains the number of rows the query changed.
+   */
```

Comment:
Document as optional field instead of MySQL-only field. Some 3rd party dialects might set this.

Reply:
The documentation doesn't say it's optional now. It says the opposite. Optionality of the type does not imply optionality of implementation in this interface.